src/core/Akka.Remote.Tests/EndpointRegistrySpec.cs

```diff
@@ -187,33 +185,64 @@ public void EndpointRegistry_should_overwrite_Gated_policy_with_Pass_on_Register
             var deadline = Deadline.Now + TimeSpan.FromMinutes(30);
             var willBeGated = 42;
 
-            reg.RegisterWritableEndpoint(address1, TestActor, willBeGated, null);
+            reg.RegisterWritableEndpoint(address1, TestActor, willBeGated);
             Assert.NotNull(reg.WritableEndpointWithPolicyFor(address1));
             Assert.True(reg.IsWritable(TestActor));
             reg.MarkAsFailed(TestActor, deadline);
             Assert.False(reg.IsWritable(TestActor));
 
             var writableUid = 43;
-            reg.RegisterWritableEndpoint(address1, TestActor, writableUid, willBeGated);
+            reg.RegisterWritableEndpoint(address1, TestActor, writableUid);
             Assert.True(reg.IsWritable(TestActor));
         }
 
         [Fact]
-        public void EndpointRegister_should_not_report_endpoint_as_writeable_if_no_Pass_policy()
+        public void EndpointRegistry_should_not_report_endpoint_as_writable_if_no_Pass_policy()
         {
             var reg = new EndpointRegistry();
             var deadline = Deadline.Now + TimeSpan.FromMinutes(30);
             
             Assert.False(reg.IsWritable(TestActor)); // no policy
 
-            reg.RegisterWritableEndpoint(address1, TestActor, 42, null);
+            reg.RegisterWritableEndpoint(address1, TestActor, 42);
             Assert.True(reg.IsWritable(TestActor)); // pass
             reg.MarkAsFailed(TestActor, deadline); 
             Assert.False(reg.IsWritable(TestActor)); // Gated
-            reg.RegisterWritableEndpoint(address1, TestActor, 43, 42); // restarted
+            reg.RegisterWritableEndpoint(address1, TestActor, 43); // restarted
             Assert.True(reg.IsWritable(TestActor)); // pass
             reg.MarkAsQuarantined(address1, 43, deadline);
-            Assert.False(reg.HasWriteableEndpointFor(address1)); // Quarantined
+            Assert.False(reg.HasWritableEndpointFor(address1)); // Quarantined
+        }
+
+        [Fact]
+        public void EndpointRegistry_should_keep_refuseUid_after_register_new_Endpoint()
```

Comment:
Validating improved `refuseUid` handling for keeping track of quarantines inside the `EndpointRegistry` - per https://github.com/akka/akka/pull/23734